### ISSUE TITLE
[ptpu] Vision SDPA kernel produces NaN/garbage output, causing image inference failure

### ISSUE THREAD

**Issue by Al2O3Al2O3** [flagos-ai/FlagGems#6123]
## Environment
- FlagGems: v5.4.0-rc0 (branch 5.4.0-rc0)
- Backend: ptpu (sunrise/曦望 S2)
- FlagTree: 0.7.0rc0+triton3.6 (FLAGTREE_BACKEND=sunrise)
- Model: Qwen3.6-27B (vision-language model)
- vLLM: 0.20.2+flagos, vllm-plugin-FL v0.2.2-rc0

## Problem
**Text inference works perfectly, but image inference produces only repeated `!` tokens (token ID 0 spam) — the classic signature of NaN/corrupted logits.**

### Working: Text inference
```bash
curl http://127.0.0.1:8000/v1/chat/completions \
  -d '{"model":"qwen","messages":[{"role":"user","content":"介绍北京"}],"max_tokens":64}'
# Output: normal Chinese text generation ✅
```

### Broken: Image inference
```bash
curl http://127.0.0.1:8000/v1/chat/completions \
  -d '{"model":"qwen","messages":[{"role":"user","content":[
    {"type":"image_url","image_url":{"url":"https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen-VL/assets/demo.jpeg"}},
    {"type":"text","text":"这张图片里有什么？"}
  ]}],"max_tokens":128}'
# Output: {"content":"!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!..."} (128 exclamation marks) ❌
# prompt_tokens=2774 (vision patches encoded correctly), but LM emits only token 0
```

## Root cause analysis
**Isolated to FlagGems vision encoder SDPA kernel on ptpu backend:**

1. **Text path is numerically healthy** — all LM kernels (rms_norm, silu_and_mul, rotary_embedding, GDN attention) produce correct outputs
2. **Vision encoder uses `_sdpa_math_kernel`** (TORCH_SDPA backend, FlagGems Triton kernel registered via aten dispatch)
3. **Image tokens are encoded** (2774 prompt tokens = 15 text + 2759 vision patches)
4. **But vision embeddings fed to LM are garbage** → LM receives NaN/corrupt input → collapses to token 0

Log shows vision encoder using FlagGems kernel:
```
(Worker_TP0) INFO mm_encoder_attention.py:230] Using AttentionBackendEnum.TORCH_SDPA for MMEncoderAttention.
warning: <unknown>:0: _sdpa_math_kernel used 144 bytes local memory!
```

## Impact
**Blocks all vision-language inference on ptpu backend.** Text-only models work fine, but multimodal (Qwen3.6, Qwen2-VL, etc.) are completely unusable.

## Expected behavior
Vision SDPA should produce numerically correct attention outputs (as it does on CUDA/other backends), allowing the LM to generate meaningful text from image content.

## Additional context
- Warm cache: first image request also failed (not a cold-kernel compile issue)
- Deterministic failure: every image request produces the same token-0 spam
- LM is healthy: text-only requests after image failures continue working correctly
- Plugin dispatch shows proper fallback: `Op 'attention_backend' fallback to 'vendor.sunrise'` for LM ops, but vision encoder directly uses FlagGems SDPA (injected via torch aten)

**This is a P0 blocker for vision workloads on ptpu.** I could not identify the specific buggy line in the SDPA implementation, only that the kernel as a whole produces corrupt output on this backend.

**Comment by Dongxu-H** [comment]
related issue link https://github.com/flagos-ai/FlagGems/issues/6562
